src/config/backends/extlinux-file.ts

```diff
@@ -0,0 +1,86 @@
+import * as _ from 'lodash';
+
+import { ConfigOptions } from './backend';
+import { ExtLinuxParseError } from '../../lib/errors';
+
+export interface ExtlinuxFile {
+	labels: Label;
+	globals: Directive;
+}
+
+export interface Label {
+	[labelName: string]: Directive;
+}
+
+export interface Directive {
+	[directive: string]: string;
+}
+
+export abstract class ConfigurableDirective {
+	// Parses the values for this directive
+	public abstract parse(directives: Directive): ConfigOptions;
+	// Return the value to be set for this directive using the provided ConfigOptions
+	public abstract value(opts: ConfigOptions, existingValue: string): string;
```

Comment:
At the risk of bike-shedding here, I'm not super fond of this method name. Something a little more applicable, such as `generate` or `compose` might be nice (although compose might be confused with docker-compose).

Reply:
I totally agree. Compose is perfect but in the off chance it can be confused with docker-compose I'll opt for generate.